static/js/factories/index.js

```diff
@@ -0,0 +1,105 @@
+// @flow
+import R from 'ramda';
+
+import {
+  STATUS_OFFERED,
+  FA_STATUS_APPROVED,
+} from '../constants';
+import type {
+  Coupon,
+  Coupons,
+} from '../flow/couponTypes';
+import type {
+  CoursePrice,
+  CoursePrices,
+  Dashboard,
+} from '../flow/dashboardTypes';
+import type {
+  Course,
+  CourseRun,
+  Program,
+} from '../flow/programTypes';
+
+const makeCounter = (): (() => number) => {
+  let gen = (function*() {
+    let i = 1;
+    while (true) {  // eslint-disable-line no-constant-condition
+      yield i;
+      i += 1;
+    }
+  })();
+  // $FlowFixMe: Flow doesn't know that this always returns a number
```

Comment:
Actually I don't think Flow can ever know about this. `next()` returns an object like `{value: 'value', done: false}`, but if the iterator ends then it returns `{done: true}` with no `value`, so it may not be a number. Flow's not smart enough to solve the halting problem :)